python/aibrix_kvcache/aibrix_kvcache/transport/rdma.py

```diff
@@ -0,0 +1,585 @@
+        elif request.gid_index >= 0:
+            logger.info(
+                "Got GID index %d specified for device %s, checking it",
+                request.gid_index,
+                device_name,
+            )
+
+            return self._select_port_by_gid_index(context, request)
+
+        # auto detect
+        return self._select_port_auto(context)
+
+    def _select_port_auto(self, context: Context) -> Status[PortAttributes]:
+        device_name = context.name
+        port_attrs: PortAttributes | None = None
+        logger.info(
+            "Trying to select port by bandwidth for device %s",
+            device_name,
+        )
+        status = self._select_port_by_bandwidth(context)
+
+        if not status.is_ok():
+            logger.warning(
+                "Failed to select port by bandwidth for device %s, "
+                "selecting the first active port",
+                device_name,
+            )
+        else:
+            port_attrs = status.get()
+
+        if port_attrs is None:
+            logger.info(
+                "Trying to select the first active port for device %s",
+                device_name,
+            )
+            status = self._select_first_active_port(context)
+            if not status.is_ok():
+                logger.warning(
+                    "No active ports found for device %s", device_name
+                )
+            else:
+                port_attrs = status.get()
+
+        if port_attrs is None:
+            logger.warning("Failed to select port for device %s", device_name)
+            return Status(StatusCodes.NOT_FOUND)
+
+        logger.info("Selected port for device %s: %s", device_name, port_attrs)
+
+        return Status.ok(port_attrs)
+
+    def _select_port_by_port_num(
+        self, context: Context, request: DeviceRequest
+    ) -> Status[PortAttributes]:
+        port_attrs = self._get_port_attributes(context, request.port_num)
+
+        if not port_attrs.is_active:
+            logger.warning(
+                "Requested port %d is not active on device %s",
+                request.port_num,
+                context.name,
+            )
+            return Status(StatusCodes.NOT_FOUND)
+
+        if port_attrs.gid_index < 0 or (
+            request.gid_index >= 0 and port_attrs.gid_index != request.gid_index
+        ):  # type: ignore
+            logger.warning(
+                "Requested port %d on device %s does not have a "
+                "valid GID index",
+                request.port_num,
+                context.name,
+            )
+            return Status(StatusCodes.NOT_FOUND)
+
+        return Status.ok(port_attrs)
+
+    def _select_port_by_gid_index(
+        self, context: Context, request: DeviceRequest
+    ) -> Status[PortAttributes]:
+        active_ports = [p for p in self._list_all_active_ports(context)]
+        if not active_ports:
+            return Status(StatusCodes.NOT_FOUND)
```

Comment:
seems most of the status code we used here are `NOT_FOUND`, any other codes to reflect the real status?